File: Signal.Delphi/IntroPeakSpectral.pas
unit IntroPeakSpectral;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  Basic3, StdCtrls, ComCtrls;

type
  TIntroPeakSpectralForm = class(TBasicForm3)
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  IntroPeakSpectralForm: TIntroPeakSpectralForm;

implementation

{$IFDEF CLR}
{$R *.nfm}
{$ELSE}
{$R *.dfm}
{$ENDIF}

procedure TIntroPeakSpectralForm.FormCreate(Sender: TObject);
begin
  inherited;
  With RichEdit1.Lines, RichEdit1 do
  begin
    Clear;
    DefAttributes.Size := 10;
    DefAttributes.Style := [fsBold];
    SelAttributes.Size := 12;

    SelAttributes.Color := RGB(51,51,153);
    Add('Frequency spectrum peak analysis:');
    Add('');
    Paragraph.FirstIndent := 10;
    Paragraph.Numbering := nsBullet;
    Add('6 methods for frequency and amplitude interpolation: '+
        'Numeric, Quadratic, Barycentric, Quinn''s First, Quinn''s second, Jain''s');
    Add('Auto find peaks in the frequency spectrum: largest, harmonics or retrace existing');
    Add('Add and delete peak marks on the fly with a dedicated TeeTool');
    Add('Peak order tracking');
    Add('Limit search for peaks within a frequency band');
    Add('Link RMS calculation of frequency bands to peak position');
    Paragraph.FirstIndent := 0;
    Paragraph.Numbering := nsNone;
  end;
end;

initialization
   RegisterClass(TIntroPeakSpectralForm);


end.

File: Signal.CPP/IntroPeakSpectral.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "IntroPeakSpectral.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TIntroPeakSpectralForm *IntroPeakSpectralForm;
//---------------------------------------------------------------------------
__fastcall TIntroPeakSpectralForm::TIntroPeakSpectralForm(TComponent* Owner)
	: TBasicForm3(Owner)
{
	RichEdit1->Lines->Clear();
	RichEdit1->DefAttributes->Size = 12;
	RichEdit1->DefAttributes->Style = RichEdit1->DefAttributes->Style << TFontStyle::fsBold;
	RichEdit1->SelAttributes->Color = (TColor) RGB(51,51,153);
	RichEdit1->SelAttributes->Size = 14;
	RichEdit1->Lines->Add("Frequency spectrum peak analysis:");
	RichEdit1->Lines->Add("");
	RichEdit1->Paragraph->Numbering = nsBullet;
	RichEdit1->Paragraph->FirstIndent = 10;
	RichEdit1->Lines->Add("6 methods for frequency and amplitude interpolation: "
		"Numeric, Quadratic, Barycentric, Quinn's First, Quinn's second, Jain's");
	RichEdit1->Lines->Add("Auto find peaks in the frequency spectrum: largest, harmonics or retrace existing");
	RichEdit1->Lines->Add("Add and delete peak marks on the fly with a dedicated TeeTool");
	RichEdit1->Lines->Add("Peak order tracking");
	RichEdit1->Lines->Add("Limit search for peaks within a frequency band");
	RichEdit1->Lines->Add("Link RMS calculation of frequency bands to peak position");
	RichEdit1->Paragraph->Numbering = nsNone;
	RichEdit1->Paragraph->FirstIndent = 0;
}
//---------------------------------------------------------------------------
